Clarify "payment" naming, purpose of the "label" field. (#406)
zcash/zips: zip-0321.rst

----------

label
   Label for an address (e.g. name of receiver). If a label is present at
   a ``paramindex``, a client rendering a payment for inspection by the user

>    SHOULD preferentially display this label instead of the associated 
>    address (the client MUST still make it possible for the user to see the 
>    actual address to which the payment will be directed.)

Reviewer: ```suggestion
   SHOULD display this label as well as the associated address.
```

Author: So, I can see this argument. But honestly, how would you ever know? If you're scanning a QR code you never see the "original" address, and if you're copying a URI you have to do some careful visual parsing to check. Furthermore, if there's malware on the device, it could just as easily change the rendering of zcash URIs to rewrite the addresses. My take is that if the user is going to validate the address, having it folded away in the UI behind an :eye: or something isn't going to change the security situation much.

Reviewer 2: So for stuff like “complex thing can be replaced without human seeing it” I recommend some kind of pictorial representation a’la SSH keys with randomart. http://users.ece.cmu.edu/~adrian/projects/validation/validation.pdf

Ultimately though of course code running as the user can access things the user can, so if that applies here, they could have the process still display the same address but use a different one in the background.

Personally I find it annoying that labels replaced the email address in google mail, for instance. But they did that for a reason, and it’s that most people find that better. I think we probably want to appeal to everyone though.

I hope that this collection of seemingly unrelated items provides enough lubrication to resolve this sticking point.
